### ISSUE TITLE
`SUM_TAXONOMY` and other summary modules produces empty output with RNASpades/Prodigal

### ISSUE THREAD

**Issue by erikrikarddaniel** [LNUc-EEMiS/metatdenovo#169]
### Description of the bug

I ran RNASpades with Prodigal and found a number of problems in the `SUM_TAXONOMY`. What alerted me was that the output file from this module was empty. I think there are similar problems (at least number 1 below) in the other modules that feed into `MERGE_TABLES` as the other two files were also empty.

1. The `orf` column of the input two tables do not have the same keys

In the work directory for `SUM_TAXONOMY` (gtdb) the counts table (`all_samples_counts.tsv.gz`) has orfs like `1_1` while the taxonomy table (`gtdb_taxonomy_classification.tsv.gz`) has orfs looking like `NODE_1000_1`. First lines of counts:

```
orf     chr     start   end     strand  length  sample  count   tpm
1_1     NODE_1  26      1420    +       1395    KA1     75      12.392417023856204
1_2     NODE_1  1461    1565    +       105     KA1     4       8.78091263404668
1_3     NODE_1  1552    1806    -       255     KA1     3       2.7117524311026515
```

And taxonomy:

```
orf     Domain  Phylum  Class   Order   Family  Genus   Species
NODE_1000_1     Bacteria         Bacteroidota    Bacteroidia     Flavobacteriales        Schleiferiaceae         UBA10364        UBA10364_sp018402665
NODE_1000_2     Bacteria         Bacteroidota    Bacteroidia     Flavobacteriales        Schleiferiaceae         UBA10364        UBA10364_sp002387615
NODE_1000_3     Bacteria         Bacteroidota    Bacteroidia     Flavobacteriales        Schleiferiaceae         UBA10364        UBA10364_sp002387615
```

2. Can there ever be more than one counts file?

I found this line confusing: `counts <- list.files(pattern = "*_counts.tsv.gz") `. This suggests that there might be more than one counts file -- is that possible? 

(If it is possible, there's a better way to deal with multiple input files: `read_tsv(Sys.glob('*_counts.tsv.gz'), show_col_types = FALSE)`: easier to understand.)

3. The `right_join()` should be a `left_join()` or an `inner_join()` I believe

There can never be orfs without counts (or, at least, we're not interested in them) so a `right_join()` does not make sense. I'm a little confused as to what the code tries to accomplish, however, so we need to discuss that @danilodileo .

4. The `count(orf)` call is odd

`count(orf)` is shorthand for `group_by(orf) %>% summarise(n = n())`. Since this is inside a `group_by(sample)` I'm not sure what you want to accomplish here.

**Comment by erikrikarddaniel** [comment]
I wasn't able to replicate this problem with the test case. `nextflow run . -profile test,docker -resume --assembler rnaspades --eukulele_db gtdb --skip_eukulele false` worked.

**Comment by erikrikarddaniel** [comment]
featureCount can take the name of the gene identifier as a command line parameter. If supplying this would help with this problem, we could have one featureCount module call per ORF caller, with different aliases. Through the aliases we could provide different command line options to featureCount via `modules.config`.

**Comment by erikrikarddaniel** [comment]
Fixed in #173
